nwbwidgets/ophys/two_photon_series.py

```diff
@@ -0,0 +1,62 @@
+import ipywidgets as widgets
+import plotly.graph_objects as go
+import plotly.express as px
+from pynwb.ophys import TwoPhotonSeries
+from tifffile import imread, TiffFile
+
+from .single_plane import SinglePlaneVisualization
+from .plane_slice import PlaneSliceVisualization
+from .volume import VolumeVisualization
+from ..base import LazyTab
+
+
+class TwoPhotonSeriesVisualization(widgets.VBox):
+    """Widget showing Image stack recorded over time from 2-photon microscope."""
+
+    def __init__(self, indexed_timeseries: TwoPhotonSeries, neurodata_vis_spec: dict):
+        super().__init__()
+
+        if indexed_timeseries.data is None:
+            if indexed_timeseries.external_file is not None:
+                path_ext_file = indexed_timeseries.external_file[0]
+                # Get Frames dimensions
+                tif = TiffFile(path_ext_file)
+                n_samples = len(tif.pages)
+                page = tif.pages[0]
+
+                def _add_fig_trace(img_fig: go.Figure, index):
+                    if self.figure is None:
+                        self.figure = go.FigureWidget(img_fig)
+                    else:
+                        self.figure.for_each_trace(lambda trace: trace.update(img_fig.data[0]))
+
+                def update_figure(index=0):
+                    # Read first frame
+                    img_fig = px.imshow(imread(path_ext_file, key=int(index)), binary_string=True)
+                    _add_fig_trace(img_fig, index)
+
+                self.slider = widgets.IntSlider(
+                    value=0, min=0, max=n_samples - 1, orientation="horizontal", description="TIFF index: "
+                )
+                self.controls = dict(slider=self.slider)
+                self.slider.observe(lambda change: update_figure(index=change.new), names="value")
+
+                update_figure()
+
+                series_name = indexed_timeseries.name
+                base_title = f"TwoPhotonSeries: {series_name}"
+                self.figure.layout.title = f"{base_title} - read from first external file"
+
+                self.children = [self.figure, self.slider]
```

Comment:
TBH this entire section could also be absorbed into a minimal child class (the only thing different is where/how the data is fetched...

Might prefer to do in separate PR though, this is more than enough as is.